src/sink/rest.rs

```diff
@@ -65,55 +78,124 @@ impl offramp::Impl for Rest {
 }
 
 impl Rest {
-    async fn flush(endpoint: &str, config: Config, payload: Vec<u8>) -> Result<u64> {
+    async fn request(
+        //endpoint: &str,
+        //config: Config,
+        payload: Vec<u8>,
+        meta: RestRequestMeta,
+    ) -> Result<(u64, Response)> {
         let start = Instant::now();
-        let mut c = if config.put {
-            surf::put(endpoint)
-        } else {
-            surf::post(endpoint)
+
+        // TODO would be nice if surf had a generic function to handle a given method
+        // also use enums (or static strings) here instead of matching on strings
+        let mut c = match meta.method.as_str() {
+            "POST" => surf::post(meta.endpoint),
+            "PUT" => surf::put(meta.endpoint),
+            "GET" => surf::get(meta.endpoint),
```

Comment:
Add support for PATCH and lets consider CONNECT and OPTIONS which are more likely to materialize in practice with LT-based REST use cases

Reply:
we should implement all of those: https://docs.rs/surf/1.0.3/surf/#functions

Reply:
Implemented as part of https://github.com/tremor-rs/tremor-runtime/pull/431/commits/6461cd38993782838017c2802bcd5e0c2ab105b2

Probably would be nice to add a wrapper method to surf itself that does request based on given http method. We can do that at a later date.

---

While on the topic, also reminder for myself to implement connection reuse for the rest offramp: https://github.com/tremor-rs/tremor-runtime/issues/220